code/game/objects/structures/fence.dm

```diff
@@ -129,13 +127,12 @@
 	return TRUE
 
 /obj/structure/fence/door/proc/toggle(mob/user)
-	open = !open
-	visible_message(span_notice("\The [user] [open ? "opens" : "closes"] \the [src]."))
+	visible_message(span_notice("\The [user] [density ? "opens" : "closes"] \the [src]."))
+	set_density(!density)
 	update_door_status()
 	playsound(src, 'sound/machines/click.ogg', 100, TRUE)
 
 /obj/structure/fence/door/proc/update_door_status()
```

Comment:
Surely, `update_door_status()` can be moved into `update_icon_state()` now, yes?